src/BizHawk.Client.Common/Controller.cs

```diff
@@ -59,11 +59,20 @@ public bool AnyInputHeld
 		public List<string> SearchBindings(string button)
 			=> _bindings.Where(b => b.Value.Contains(button)).Select(static b => b.Key).ToList();
 
-		// Searches bindings for the controller and returns true if this binding is mapped somewhere in this controller
-		public bool HasBinding(string button) =>
-			_bindings
+		/// <summary>
+		/// Checks if the given button combination would be a controller input.
+		/// This means Shift+A will return true if an input is bound to either A or Shift+A.
+		/// </summary>
+		public bool HasBinding(string button)
+		{
+			HashSet<string> buttons = button.Split('+').ToHashSet();
+			return _bindings
 				.SelectMany(kvp => kvp.Value)
-				.Any(boundButton => boundButton == button);
+				.Any((boundCombination) => {
+					HashSet<string> boundButtons = boundCombination.Split('+').ToHashSet();
+					return boundButtons.IsSubsetOf(buttons);
+				});
+		}
```

Comment:
I don't think this logic is fully correct. When an input is bound to e.g. `Ctrl+E` and a hotkey is bound to `Ctrl+Shift+E` and "Input overrides Hotkeys" is active, pressing `Ctrl+Shift+E` will do nothing.

Reply:
In my testing, `Ctrl+Shift+E` will press the controller input and not trigger the hotkey, which is the expected behavior.
Inputs work even when pressing extra modifier keys because (from Discord)
> If we didn't consider Shift+A as also A then you could not press whatever is bound to Shift and then whatever is bound to A while still holding shift.

Reply:
Have you tested this with `Input overrides Hotkeys`? Because I get nothing.